daemon/SecretsImpl/secrets.cpp

```diff
@@ -2586,3 +2594,23 @@ void Daemon::ApiImpl::SecretsRequestQueue::handleFinishedRequest(
     }
 }
 
+void Daemon::ApiImpl::SecretsRequestQueue::dealWithData() const
+{
+    // Find secrets directory
+    const static QString systemDataDirPath(QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + "/system/");
```

Comment:
I think this directory is built somewhere else in the code (like the place where to create the databases). It would be better to have a method that return this path, so in case of a change, we need to change only once and don't have to grep in the code to find all the places to change.